packages/styled-components/src/constructors/constructWithOptions.ts

```diff
@@ -69,12 +70,15 @@ export interface Styled<
   ) => Styled<
     R,
     PrivateResolvedTarget,
-    PrivateResolvedTarget extends KnownTarget
-      ? Substitute<
-          Substitute<OuterProps, React.ComponentPropsWithRef<PrivateResolvedTarget>>,
-          Props
-        >
-      : PrivateMergedProps,
+    Optional<
+      PrivateResolvedTarget extends KnownTarget
+        ? Substitute<
+            Substitute<OuterProps, React.ComponentPropsWithRef<PrivateResolvedTarget>>,
```

Comment:
Using ComponentPropsWithRef after importing it with import type { ComponentPropsWithRef } from 'react'; would improve readability, I think.